ovos_plugin_manager/templates/audio.py

```diff
@@ -58,52 +63,143 @@ def runtime_requirements(self):
 
-    # methods below deprecated and handled by OCP directly
-    # playlists are no longer managed plugin side
-    # this is just a compat layer forwarding commands to OCP
+    @property
+    def playback_time(self):
+        return 0
+
+    @abstractmethod
+    def supported_uris(self):
+        """List of supported uri types.
+
+        Returns:
+            list: Supported uri's
+        """
+
+    @abstractmethod
     def clear_list(self):
         """Clear playlist."""
-        msg = Message('ovos.common_play.playlist.clear')
-        self.bus.emit(msg)
 
+    @abstractmethod
     def add_list(self, tracks):
         """Add tracks to backend's playlist.
 
         Arguments:
             tracks (list): list of tracks.
         """
-        tracks = tracks or []
-        if isinstance(tracks, (str, tuple)):
-            tracks = [tracks]
-        elif not isinstance(tracks, list):
-            raise ValueError
-        tracks = [self._uri2meta(t) for t in tracks]
-        msg = Message('ovos.common_play.playlist.queue',
-                      {'tracks': tracks})
-        self.bus.emit(msg)
-
-    @staticmethod
-    def _uri2meta(uri):
-        if isinstance(uri, list):
-            uri = uri[0]
-        try:
-            from ovos_ocp_files_plugin.plugin import OCPFilesMetadataExtractor
-            return OCPFilesMetadataExtractor.extract_metadata(uri)
-        except:
-            meta = {"uri": uri,
-                    "skill_id": "mycroft.audio_interface",
-                    "playback": PlaybackType.AUDIO,  # TODO mime type check
-                    "status": TrackState.QUEUED_AUDIO,
-                    }
-        return meta
+
+    @abstractmethod
+    def play(self, repeat=False):
+        """Start playback.
+
+        Starts playing the first track in the playlist and will contiune
+        until all tracks have been played.
+
+        Arguments:
+            repeat (bool): Repeat playlist, defaults to False
+        """
+
+    @abstractmethod
+    def stop(self):
+        """Stop playback.
+
+        Stops the current playback.
+
+        Returns:
+            bool: True if playback was stopped, otherwise False
+        """
+
+    def set_track_start_callback(self, callback_func):
+        """Register callback on track start.
+
+        This method should be called as each track in a playlist is started.
+        """
+        self._track_start_callback = callback_func
+
+    def pause(self):
+        """Pause playback.
+
+        Stops playback but may be resumed at the exact position the pause
+        occured.
+        """
+
+    def resume(self):
+        """Resume paused playback.
+
+        Resumes playback after being paused.
+        """
 
     def next(self):
         """Skip to next track in playlist."""
-        self.bus.emit(Message("ovos.common_play.next"))
 
     def previous(self):
         """Skip to previous track in playlist."""
-        self.bus.emit(Message("ovos.common_play.previous"))
```

Comment:
this one should also be left there with default code

Reply:
reverted with try/except import